src/lib/teuthologyAPI.ts

```diff
@@ -3,28 +3,46 @@ import { useQuery } from "@tanstack/react-query";
 import { Cookies } from "react-cookie";
 import type { UseQueryResult } from "@tanstack/react-query";
 
-const TEUTHOLOGY_API_SERVER = 
+const TEUTHOLOGY_API_SERVER =
     import.meta.env.VITE_TEUTHOLOGY_API || "";
 const GH_USER_COOKIE = "GH_USER";
 
-function getURL(relativeURL: URL|string): string {
-    if ( ! TEUTHOLOGY_API_SERVER ) return "";
+function getURL(relativeURL: URL | string): string {
+    if (!TEUTHOLOGY_API_SERVER) return "";
     return new URL(relativeURL, TEUTHOLOGY_API_SERVER).toString();
 }
 
 function doLogin() {
     const url = getURL("/login/");
-    if ( url ) window.location.href = url;
+    if (url) window.location.href = url;
 }
 
 function doLogout() {
     const cookies = new Cookies();
     cookies.remove(GH_USER_COOKIE);
-    
+
     const url = getURL("/logout/");
     window.location.href = url;
 }
 
+async function doSchedule(commandValue: any) {
+    const url = getURL("/suite?logs=true");
+    if (commandValue['--user'] != useUserData().get("username")) {
+        console.log("Error: --user doesn't match username of current logged in account");
+        return false;
+    }
+    return await axios.post(url, commandValue, {
```

Comment:
@kamoltat
In order to kill a run scheduled by pulpito-ng's schedule feature, teuthology would look for owner (which is set by `"--owner"` parameter). If no `"--owner"` is passed in teuthology-suite command, teuthology sets it as `scheduled_<machine username>@<machine hostname>`. 
In teuthology-api and pulpito-ng setup, I think it'll work out well if we set `commandValue['--owner'] = github_username` on pulpito-ng before sending request to t-api. This way we would be able to recognise the owner as github username when we try to kill the run later. 

Also, I think we can skip "--user" checks entirely since we merged https://github.com/ceph/teuthology-api/pull/47/files#diff-d64c10972091566c2a44e7e643a8e41b7a4781efb8cddb9516b0cfe320ff648f in which we set "--user" as github username in teuthology-api.

Reply:
Thank you, new commit should address this.